core/src/main/java/com/whylogs/core/ColumnProfile.java

```diff
@@ -31,28 +33,52 @@
 public class ColumnProfile {
   public static final int FREQUENT_MAX_LG_K = 7;
   private static final int CARDINALITY_LG_K = 12;
+  private static Map<String, String> ENV;
+  private static ImmutableList<String> nullEnvStrs;
 
   @NonNull private final String columnName;
   @NonNull private final CountersTracker counters;
   @NonNull private final SchemaTracker schemaTracker;
   @NonNull private final NumberTracker numberTracker;
   @NonNull private final ItemsSketch<String> frequentItems;
   @NonNull private final HllSketch cardinalityTracker;
+  private final ImmutableList<String> nullStrs;
+
+  static synchronized ImmutableList<String> nullStrsFromEnv() {
```

Comment:
in general it's better to use `volatile` on the variable (specifically NULL_STR_ENVS) if the code path is idempotent (i.e. race conditions won't lead to different results)